Minio/DataModel/BucketOperationsArgs.cs

```diff
@@ -0,0 +1,156 @@
+/*
+ * MinIO .NET Library for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using Minio.DataModel;
+using RestSharp;
+
+namespace Minio
+{
+    public class BucketExistsArgs: BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new BucketExistsArgs WithBucket(string bucket)
+        {
+            this.BucketName = bucket;
+            return this;
+        }
+
+        public new BucketExistsArgs WithLocation(string loc)
+        {
+            this.Location = loc;
+            return this;
+        }
+        public BucketExistsArgs WithRegion(string reg)
+        {
+            this.Region = reg;
+            return this;
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.HEAD, authenticator, this.BucketName, this.Secure, this.Region );
+        }
+    }
+
+    public class RemoveBucketArgs : BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new RemoveBucketArgs WithBucket(string bucket)
+        {
+            this.BucketName = bucket;
+            return this;
+        }
+
+        public new RemoveBucketArgs WithLocation(string loc)
+        {
+            this.Location = loc;
+            return this;
+        }
+        public RemoveBucketArgs WithRegion(string reg)
+        {
+            this.Region = reg;
+            return this;
+        }
+        public new RestRequest GetRequest(string baseUrl, RestSharp.Authenticators.IAuthenticator authenticator)
+        {
+            return RequestUtil.CreateRequest(baseUrl, Method.DELETE, authenticator, this.BucketName, this.Secure, this.Region);
+        }
+
+    }
+
+    public class ListObjectsArgs : BucketArgs
+    {
+        public new void Validate()
+        {
+            utils.ValidateBucketName(this.BucketName);
+        }
+        public new ListObjectsArgs WithLocation(string loc)
+        {
+            this.Location = loc;
+            return this;
+        }
+        public ListObjectsArgs WithRegion(string reg)
+        {
+            this.Region = reg;
+            return this;
+        }
```

Comment:
All the above methods will be inherited from BucketArgs, can be removed.

Reply:
Changed to call base method with `new`.